src/main/java/org/sonatype/repository/conan/internal/security/ConanContentValidator.java

```diff
@@ -0,0 +1,52 @@
+package org.sonatype.repository.conan.internal.security;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.mime.MimeRulesSource;
+import org.sonatype.nexus.repository.storage.ContentValidator;
+import org.sonatype.nexus.repository.storage.DefaultContentValidator;
+import org.sonatype.nexus.repository.view.ContentTypes;
+import org.sonatype.repository.conan.internal.ConanFormat;
+
+import com.google.common.base.Supplier;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * @since 0.0.2
+ */
+@Named(ConanFormat.NAME)
+@Singleton
+public class ConanContentValidator
+    implements ContentValidator
+{
+  private final DefaultContentValidator defaultContentValidator;
+
+  @Inject
+  public ConanContentValidator(final DefaultContentValidator defaultContentValidator) {
+    this.defaultContentValidator = checkNotNull(defaultContentValidator);
+  }
+
+  @Nonnull
+  @Override
+  public String determineContentType(final boolean strictContentTypeValidation,
+                                     final Supplier<InputStream> contentSupplier,
+                                     @Nullable final MimeRulesSource mimeRulesSource,
+                                     @Nullable final String contentName,
+                                     @Nullable final String declaredContentType) throws IOException
+  {
+    if (contentName.endsWith(".tgz")) {
```

Comment:
If `contentName` is `@Nullable` won't this potentially blow up?

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/6/commits/e4f4649b29f5d7fe2b926eaccfb2445286c3adca